src/Guides/RestructuredText/Parser/DocumentParser.php

```diff
@@ -157,6 +155,21 @@ private function init(): void
     private function setState(string $state): void
     {
         $this->state = $state;
+        $this->subparser = null;
```

Comment:
That would be the case if this was a pushdown automaton, for example, but this is a simple finite state machine that does not have a stack of actions through which it can nest